en/06_Project_Governance/08_Public_API.md

```diff
@@ -19,6 +19,8 @@ These are explicitly included in our definition of public API (unless excluded b
 * **global** functions, constants, and variables
 * namespaces, classes, interfaces, enums, and traits
 * public and protected scope (including methods, properties and constants)
+* signatures of functions/methods that are considered public API
+  * including typehints, parameters, and parameter names
```

Comment:
Looks good, but I have small question.
I'm not sure, but I'll ask. Should we include the return type and possible exceptions thrown in this list? Not as part of the method signature, but as part of the method contract.

Reply:
The return type _is_ part of the method signature. The only reason I explicitly mention parameters is because changing parameter _names_ isn't widely known as a breaking change (it only became so with PHP 8.0 introducing named arguments) and because people routinely try to add optional parameters to pre-existing methods. The idea is to stop people doing that.

Exceptions is a bit of a weird one, because _most_ of the time changing exceptions would be a breaking change, but there might be cases where something currently throws an exception and we want to _stop_ throwing that exception, e.g. if the exception implies a feature hasn't been implemented, but we decide to implement it. We'd obviously need to remove that exception.